### PR TITLE
stft: move towards always returning complex

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #72882

For `stft` this makes all cases where `return_complex` default to
`False` into an error and adds a warning when `return_complex=False`
is passed explicitly.

For `istft` this raises an error if the input is not a complex tensor.

### CODE REVIEW

File: torch/csrc/jit/operator_upgraders/upgraders_entry.cpp [hunk 1]
```diff
@@ -15,6 +15,27 @@ namespace torch {
...
+              onesided: Optional[bool] = None, length: Optional[int] = None,
+              return_complex: bool = False) -> Tensor:
+  if not self.is_complex():
+    self = torch.view_as_complex(self.contiguous())
+  return torch.istft(self, n_fft=n_fft, hop_length=hop_length, win_length=win_length,
+                     window=window, center=center, normalized=normalized, onesided=onesided,
+                     length=length, return_complex=return_complex)
+)SCRIPT"},
+    {"stft_0_10", R"SCRIPT(
+def stft_0_10(self: Tensor, n_fft: int, hop_length: Optional[int]=None, win_length: Optional[int]=None,
```
[the remark this answers is not in the document]
Comment (mruberry) [reply torch/csrc/jit/operator_upgraders/upgraders_entry.cpp:31]: @tugsbayasgalan should know what to do

[the remark this answers is not in the document]
Comment (peterbell10) [reply torch/csrc/jit/operator_upgraders/upgraders_entry.cpp:31]: #71993 removes the `istft` python wrapper but is blocked by a JIT bug (#71994). `stft` depends on `torch.nn.functional.pad` which is implemented in python.

[the remark this answers is not in the document]
Comment (tugsbayasgalan) [reply torch/csrc/jit/operator_upgraders/upgraders_entry.cpp:31]: We currently iterate over all nodes in the graph and replace the aten operators with upgraders when necessary. I am wondering if we can be little smarter in that pass that would dig deeper when we encounter `prim::CallFunction`. I am not sure how feasible that is @eellison @gmagogsfm. Alternatively, fixing the bug @anjali411 mentioned to unblock the  moving to C++ might be the way to go. What do you guys think?

**Comment by peterbell10** [comment]
Sure thing. The actual code shouldn't need changed, but I'm still waiting on #73434 which is blocked by an FC period that ends this week,

**Comment by anjali411** [comment]
@peterbell10 we should merge this PR right?

**Comment by peterbell10** [comment]
I should have said, the deprecations have been re-submitted as #86724 and #86628 which have been merged.

**Pull request closed without merging**